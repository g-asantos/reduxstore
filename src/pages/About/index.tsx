import React from 'react';

import {
  AboutTitle,
  Container,
  TextContainer,
  Image,
  AboutText,
} from './styles';
import logo from '../../assets/undraw_Online_page_re_lhgx.svg';
import Header from '../../components/Header';

const About: React.FC = () => {
  return (
    <>
      <Header />
      <Container>
        <TextContainer>
          <AboutTitle>ReduxStore</AboutTitle>
          <AboutText>
            This is a project created by Guilherme Azevedo dos Santos, with the
            intent to further my knowledge about Redux and Redux Saga. It was
            built using ReactJS, Redux, Redux-Saga, Styled Components and
            Prismic. To check it in more detail, go{' '}
            <a href="https://github.com/g-asantos/reduxstore">here</a>
          </AboutText>
        </TextContainer>

        <Image src={logo} />
      </Container>
    </>
  );
};

export default About;
