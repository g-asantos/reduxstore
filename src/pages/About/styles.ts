import styled from 'styled-components';

export const Container = styled.div`
  display: grid;
  grid-template-columns: 1fr 1fr;
`;

export const TextContainer = styled.div`
  display: flex;
  flex-direction: column;
  align-items: center;
  width: 100vh;
  height: 100vh;
`;

export const AboutTitle = styled.h1`
  font-weight: 600;
  margin-top: 100px;
  margin-bottom: 20px;
`;

export const AboutText = styled.p`
  text-align: justify;
  width: 60vh;
`;

export const Image = styled.img`
  width: 50vh;
  margin-top: 160px;
  margin-left: 100px;
`;
