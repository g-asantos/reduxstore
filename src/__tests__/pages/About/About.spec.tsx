import React from 'react';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import configureStore from 'redux-mock-store';
import About from '../../../pages/About';

const mockStore = configureStore([]);

describe('About page', () => {
  let store;
  let component: any;

  beforeEach(() => {
    store = mockStore({
      cart: {
        items: [],
        failedStockCheck: [],
      },
      details: '',
    });

    component = renderer.create(
      <Provider store={store}>
        <BrowserRouter>
          <About />
        </BrowserRouter>
      </Provider>,
    );
  });

  test('snapshot renders', () => {
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
