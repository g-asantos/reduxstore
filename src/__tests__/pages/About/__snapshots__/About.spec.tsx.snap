// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`About page snapshot renders 1`] = `
Array [
  <div
    className="sc-dlMBXb kGopYH"
  >
    <div
      className="sc-kfYqjs hiNhpb"
    >
      <div
        className="sc-bCwgka hgwrqX"
      >
        <a
          href="/"
          onClick={[Function]}
          style={
            Object {
              "display": "flex",
              "textDecoration": "none",
            }
          }
        >
          <div
            className="sc-iwaifL lffmvg"
          >
            Shop
          </div>
        </a>
        <a
          href="/about"
          onClick={[Function]}
          style={
            Object {
              "display": "flex",
              "textDecoration": "none",
            }
          }
        >
          <div
            className="sc-iwaifL lffmvg"
          >
            About
          </div>
        </a>
      </div>
      <button
        className="sc-fKgIGh jvxZpl"
      >
        <a
          href="/"
          onClick={[Function]}
          style={
            Object {
              "display": "flex",
              "textDecoration": "none",
            }
          }
        >
          <svg
            fill="currentColor"
            height={26}
            role="img"
            size={26}
            stroke="currentColor"
            strokeWidth="0"
            style={
              Object {
                "color": undefined,
              }
            }
            viewBox="0 0 24 24"
            width={26}
            xmlns="http://www.w3.org/2000/svg"
          >
            <title />
            <path
              d="M16.633 16.504c.869-.075 1.543-.84 1.499-1.754-.046-.914-.795-1.648-1.708-1.648h-.061c-.943.031-1.678.824-1.648 1.769.03.479.226.869.494 1.153-1.048 2.038-2.621 3.536-5.004 4.795-1.603.838-3.296 1.154-4.944.929-1.378-.194-2.456-.81-3.116-1.798-.988-1.499-1.078-3.116-.255-4.734.601-1.169 1.499-2.023 2.099-2.443-.15-.389-.33-1.048-.42-1.542-4.436 3.177-3.985 7.521-2.637 9.574 1.004 1.498 3.057 2.456 5.304 2.456.599 0 1.229-.044 1.843-.194 3.896-.749 6.847-3.086 8.54-6.532l.014-.031zM21.981 12.758c-2.321-2.727-5.738-4.225-9.634-4.225h-.51c-.253-.554-.837-.899-1.497-.899h-.045c-.943 0-1.678.81-1.647 1.753.03.898.794 1.648 1.708 1.648h.074c.675-.03 1.259-.45 1.498-1.049h.555c2.309 0 4.495.674 6.488 1.992 1.527 1.004 2.622 2.322 3.236 3.896.538 1.288.509 2.547-.045 3.597-.854 1.647-2.293 2.517-4.195 2.517-1.199 0-2.367-.375-2.967-.644-.359.298-.959.793-1.394 1.093 1.318.598 2.652.943 3.94.943 2.922 0 5.093-1.647 5.918-3.236.898-1.798.824-4.824-1.469-7.416l-.014.03zM6.49 17.042c.029.899.793 1.648 1.708 1.648h.06c.959-.03 1.693-.823 1.648-1.768 0-.899-.779-1.647-1.693-1.647h-.061c-.06 0-.149 0-.225.029-1.243-2.098-1.768-4.346-1.572-6.771.119-1.828.719-3.417 1.797-4.735.899-1.124 2.592-1.679 3.746-1.708 3.236-.061 4.585 3.971 4.689 5.574l1.498.449c-.345-4.914-3.4-7.492-6.322-7.492-2.742 0-5.273 1.993-6.293 4.915-1.393 3.896-.479 7.641 1.229 10.638-.149.195-.239.539-.209.868z"
            />
          </svg>
          <div
            className="sc-iwaifL lffmvg"
          >
            ReduxStore
          </div>
        </a>
      </button>
      <div
        className="sc-cxNIbT ejzGpj"
      >
        <button
          className="sc-lmgQde jsaQOh"
          onClick={[Function]}
        >
          <svg
            fill="none"
            height={26}
            size={26}
            stroke="currentColor"
            strokeLinecap="round"
            strokeLinejoin="round"
            strokeWidth="2"
            style={
              Object {
                "color": undefined,
              }
            }
            viewBox="0 0 24 24"
            width={26}
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M6 2L3 6v14a2 2 0 0 0 2 2h14a2 2 0 0 0 2-2V6l-3-4z"
            />
            <line
              x1="3"
              x2="21"
              y1="6"
              y2="6"
            />
            <path
              d="M16 10a4 4 0 0 1-8 0"
            />
          </svg>
          <span
            className="sc-iJCRLp ijiNjl"
          >
            0
          </span>
        </button>
      </div>
    </div>
    
  </div>,
  <div
    className="sc-bdnylx bBbcZw"
  >
    <div
      className="sc-gtssRu ezLaqA"
    >
      <h1
        className="sc-dlnjPT dJWhpT"
      >
        ReduxStore
      </h1>
      <p
        className="sc-hKFyIo cngpfo"
      >
        This is a project created by Guilherme Azevedo dos Santos, with the intent to further my knowledge about Redux and Redux Saga. It was built using ReactJS, Redux, Redux-Saga, Styled Components and Prismic. To check it in more detail, go
         
        <a
          href="https://github.com/g-asantos/reduxstore"
        >
          here
        </a>
      </p>
    </div>
    <img
      className="sc-eCApGN dTHfCG"
      src="undraw_Online_page_re_lhgx.svg"
    />
  </div>,
]
`;
